torchtitan/config/job_config.py

```diff
@@ -555,6 +555,10 @@ class ActivationCheckpoint:
     Whether to stop recomputing early when all activations have already been
     rematerialized.
     """
+    budget_ac_option: float = 1.0
+    """
+    Limit memory budget for AC from 0-100%. Works only with compile enabled.
```

Comment:
For the doc here, we could do something like:
```
When mode is set to "memory_budget", this value determines how much
partitioner in the compiler should trade off compute for memory.
0.0 corresponds to the activation memory from applying
activation checkpointing to the full compiled region, and 1.0 corresponds to
the activation memory from the default runtime-optimized strategy. 
```
(taken from `torch/_functorch/config.py`)